src/app/core/services/label.service.ts

```diff
@@ -69,6 +70,57 @@ export class LabelService {
     return color;
   }
 
+  /**
+   * Verifies that each label of its corresponding type exists in the current
+   * repository and creates the label if it does not exist.
+   */
+  private createMissingLabels(): void {
+    for (const labelType of Object.keys(ISSUE_LABELS)) {
+      const compulsoryLabels: string[] = ISSUE_LABELS[labelType];
+
+      compulsoryLabels.forEach(labelName => {
+        if (this.labelExists(labelType, labelName)) {
+          return;
+        }
+
+        const colour: string = this.getRandomLabelColour();
+
+        // Update Local Data.
+        this.getLabelList(labelType).push({labelValue: labelName, labelColor: colour});
+        this.labelColorMap.set(labelName, colour);
+
+        // Update Remote Data.
+        this.githubService.createLabel(this.getFormattedLabelName(labelType, labelName), colour);
```

Comment:
What happens if the creation of the label fails?

Reply:
Should I add a retry till success?

Reply:
Or to throw and error and prompt a re-login?

Reply:
If I am not wrong, if we make a request to tag an issue with an unknown label, GitHub will automatically create the label for us (With a default color of grey). So I think this creation of labels is so critical that we need to retry/re-login. Maybe we can have a default color scheme that we can fall back to?

Reply:
Hmm the issue is like the label retrieval is only done once on log in and we don't prompt the server for data any longer. So if lets say the colour is grey (creating the label wasn't successful) we won't know until the subsequent login. If github auto creates the label and we don't want the colour to be grey then perhaps I can do a manual check for if the colour is grey then update it to a random colour on login?

Reply:
Sure! Sounds good :)

Reply:
Sure! Its done.